src/stores/connectors/eth1/destroy.go

```diff
@@ -19,26 +19,30 @@ func (c Connector) Destroy(ctx context.Context, addr ethcommon.Address) error {
 		return err
 	}
 
-	acc, err := c.db.GetDeleted(ctx, addr.Hex())
-	if err != nil {
-		return err
+	_, err = c.Get(ctx, addr)
+	if err == nil {
+		logger.WithError(err).Error("try to destroy an exiting eth1Account, must be deleted first")
+		return errors.StatusConflictError("account %s must be deleted first", addr.Hex())
 	}
 
-	err = c.db.RunInTransaction(ctx, func(dbtx database.ETH1Accounts) error {
-		err = dbtx.Purge(ctx, addr.Hex())
+	acc, err := c.db.GetDeleted(ctx, addr.Hex())
+	if err == nil {
```

Comment:
Previous behaviour was okay 
```
HTTP DELETE
When you invoke N similar DELETE requests, first request will delete the resource and response will be 200 (OK) or 204 (No Content). Other N-1 requests will return 404 (Not Found). Clearly, the response is different from first request, but there is no change of state for any resource on server side because original resource is already deleted. So, DELETE is idempotent.
```